docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,307 @@
+which typically includes:
+* A minimum of 5×3 question-and-answer pairs.
+* Reference documents stored in Git.
+
+The goal is not to generate training data for InstructLab-trained models but to utilize the documents for RAG 
+workflows with pre-tuned models.
+
+#### Usage
+The generated artifacts can later be used to generete and ingest the embeddings into a vector database.
+
+### 2.3 Document Processing Pipeline Options
+
+
+| Option Description | Default Value | CLI Flag | Environment Variable |
+|--------------------|---------------|----------|----------------------|
+| Base directories where models are stored. | `$HOME/.cache/instructlab/models`  | `--model-dir` | `ILAB_MODEL_DIR` |
+| Name of the embedding model. | **TBD** | `--embedding-model` | `ILAB_EMBEDDING_MODEL_NAME` |
+
+### 2.4 Embedding Ingestion Pipeline
+The proposal is to add an `ingest` sub-command to the `data` command group:
+```
+ilab data ingest /path/to/docs/folder
+```
+
+#### Working Assumption
+The documents at the specified path have already been processed using the `data process` command or an equivalent method
+(see [Getting Started with Knowledge Contributions][ilab-knowledge]).
+
+#### Command Purpose
+Generate the embeddings from the pre-processed documents at */path/to/docs/folder* folder and store them in the
+configured vector database.
+
+**Notes**:
+* To ensure consistency and avoid issues with document versioning or outdated embeddings, the ingested collection will be cleared before execution. 
+  This ensures it contains only the embeddings generated from the most recent run.
+
+### Why We Need It
+To populate embedding vector stores with pre-processed information that can be used at chat inference time.
+
+#### Supported Databases
+The command may support various vector database types. A default configuration will align with the selected 
+InstructLab technology stack.
+
+#### Usage
+The generated embeddings can later be retrieved from a vector database and converted to text, enriching the 
+context for RAG-based chat pipelines.
+
+### 2.5 Embedding Ingestion Pipeline Options
+
+| Option Description | Default Value | CLI Flag | Environment Variable |
+|--------------------|---------------|----------|----------------------|
+| Vector DB implementation, one of: `milvuslite`, **TBD** | `milvuslite` | `--vectordb-type` | `ILAB_VECTORDB_TYPE` |
+| Vector DB service URI. | `./rag-output.db` | `--vectordb-uri` | `ILAB_VECTORDB_URI` |
+| Vector DB collection name. | `IlabEmbeddings` | `--vectordb-collection-name` | `ILAB_VECTORDB_COLLECTION_NAME` |
+| Base directories where models are stored. | `$HOME/.cache/instructlab/models`  | `--model-dir` | `ILAB_MODEL_DIR` |
+| Name of the embedding model. | **TBD** | `--embedding-model` | `ILAB_EMBEDDING_MODEL_NAME` |
+
+### 2.6 RAG Chat Pipeline Command
+The proposal is to add a `--rag` flag to the `model chat` command, like:
```

Comment:
Since I asked the clarifying question in a meeting, let's clearly say that if `--rag` is for overwriting file-based configuration and that if the config file has RAG enabled, the user does not need to pass this parameter.